crates/sui-indexer-alt-graphql/src/api/types/transaction/mod.rs

```diff
@@ -218,7 +221,23 @@ impl Transaction {
         };
 
         let tx_bounds = tx_bounds(ctx, &cp_bounds, global_tx_hi).await?;
-        let tx_digest_keys = tx_unfiltered(&tx_bounds, &page);
+
+        // Inclusive cursor bounds
+        let pg_lo = page
+            .after()
+            .map_or(tx_bounds.start, |cursor| cursor.max(tx_bounds.start));
+        let pg_hi = page
+            .before()
+            .map(|cursor: &JsonCursor<u64>| cursor.saturating_add(1))
+            .map_or(tx_bounds.end, |cursor| cursor.min(tx_bounds.end));
+
+        let tx_bounds = pg_lo..pg_hi;
```

Comment:
I will leave it with @henryachen to say what the best way to get these bounds is now.